backend/app/services/internal/auth_service.py

```
@@ -1,76 +1,88 @@
             raise BadRequestException(message="Email already registered")
-        
-        password_hash = get_password_hash(user_data.password)
 
+        password_hash = get_password_hash(user_data.password)
         try:
-            user = self.user_repo.create_user(user_data, password_hash)
-            return UserResponse.model_validate(user)
+            user_orm = self.user_repo.create_user(db, user_data, password_hash)
+            return orm_to_pydantic(user_orm, UserResponse)
         except ValueError as e:
             raise BadRequestException(message=str(e))
```

Ruff (0.14.13)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)